### PR TITLE
Use a common prefix to namespace resources created in a single scenario

### PR DESCRIPTION
**Pull request by wjbuys**
This changes the InputCreator to prefix the same UUID to all resources
it creates, and sets up the DataServiceSteps as a DirtiesContext. The
effect of this is that all resources that are created in the same
scenario run will have the same prefix, and that prefix will be
different from all other scenario runs.

This lets us run the tests more than once, without needing to clean up
data.

#### Testing
Apart from `gradle check`, the `gradle integ-tests:cucumberTest` target passes except for the following:

```
Feature: Create environment

  Scenario: Create an environment                  # /Users/buys/workspaces/blox/blox/integ-tests/src/cucumberTest/resources/features/dataservice/create_environment.feature:5
    When I create an environment                   # DataServiceSteps.java:47
    Then the created environment response is valid # DataServiceSteps.java:53

  Scenario: Create an environment that already exists                                          # /Users/buys/workspaces/blox/blox/integ-tests/src/cucumberTest/resources/features/dataservice/create_environment.feature:9
    Given I create an environment named "test" in the cluster "testCluster"                    # DataServiceSteps.java:98
    When I try to create another environment with the name "test" in the cluster "testCluster" # DataServiceSteps.java:105
    Then there should be a ResourceExistsException thrown                                      # DataServiceSteps.java:188
      org.junit.ComparisonFailure: expected:<R[esourceExists]Exception> but was:<R[untime]Exception>
        at org.junit.Assert.assertEquals(Assert.java:115)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at cucumber.steps.dataservice.DataServiceSteps.lambda$new$13(DataServiceSteps.java:192)
        at ✽.Then there should be a ResourceExistsException thrown(/Users/buys/workspaces/blox/blox/integ-tests/src/cucumberTest/resources/features/dataservice/create_environment.feature:12)

    And the resourceType is "environment"                                                      # DataServiceSteps.java:195
    And the resourceId contains "test"                                                         # DataServiceSteps.java:214
```

This still fails because of #315 (exceptions from JSON-RPC aren't mapped correctly), and can be tackled as part of that issue.

### CODE REVIEW

File: integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java [hunk 1]
```diff
@@ -20,95 +20,100 @@
...
-        .taskDefinition(TASK_DEFINITION_ARN)
+        .environmentId(id)
+        .role(getRoleArn())
+        .taskDefinition(getTaskDefinitionArn())
         .environmentType(EnvironmentType.Daemon)
         .deploymentMethod("ReplaceAfterTerminate")
         .build();
   }
 
+  private EnvironmentId environmentId(final String environmentName, final String cluster) {
```
Comment (xjhe) [inline integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java:79]: It's clearer to name this method as "createEnvironmentId" given that you're generating a new ID with the name and cluster?

File: integ-tests/src/cucumberTest/java/cucumber/steps/dataservice/DataServiceSteps.java [hunk 2]
```diff
@@ -143,7 +145,7 @@ public DataServiceSteps() {
           final EnvironmentId environmentId = getEnvironmentIdFromCreatedEnvironment();
           dataServiceWrapper.updateEnvironment(
               inputCreator.updateEnvironmentRequestWithNewCluster(
-                  environmentId.getEnvironmentName(), newCluster));
+                  environmentId.getEnvironmentName(), inputCreator.prefixName(newCluster)));
```
Comment (xjhe) [inline integ-tests/src/cucumberTest/java/cucumber/steps/dataservice/DataServiceSteps.java:148]: This is not in the scope of this PR, but should we change this update into "I update the created environment with new task definition" ? For the purpose of testing, updating the cluster name should be fine. But I'm not sure in reality will there be a use case of updating the cluster.

File: integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java [hunk 3]
```diff
@@ -20,95 +20,100 @@
 import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.CreateEnvironmentRequest;
 import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.DeleteEnvironmentRequest;
 import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.DescribeEnvironmentRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.ListClustersRequest;
 import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.UpdateEnvironmentRequest;
-
+import java.util.Collections;
+import java.util.StringJoiner;
 import java.util.UUID;
+import lombok.Getter;
```
Comment (xjhe) [inline integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java:27]: Are we using Getter/Setter here?

File: integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java [hunk 4]
```diff
@@ -20,95 +20,100 @@
...
-  private static final int ACCOUNT_ID_SIZE = 12;
-  private static final String ACCOUNT_ID = generateAccountId();
-  private static final String TASK_DEFINITION_ARN =
-      "arn:aws:ecs:us-east-1:" + ACCOUNT_ID + ":task-definition/sleep";
-  private static final String ROLE_ARN = "arn:aws:iam::" + ACCOUNT_ID + ":role/testRole";
-  private static final String CLUSTER_NAME_PREFIX = "cluster";
+  private static final String DEFAULT_ACCOUNT_ID = "123456789012";
+  private static final String DEFAULT_CLUSTER_NAME = "Cluster";
+  private static final String DEFAULT_ENVIRONMENT_NAME = "Environment";
+  private final String sharedPrefix = "blox-integ-tests";
```
Comment (xjhe) [inline integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java]: Should we make sharedPrefix static as well?

File: integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java (hunk 3 above)
Comment (wjbuys) [reply integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java:27]: Should be on the `accountId` field, yep. Didn't want to apply it to the whole class.

File: integ-tests/src/cucumberTest/java/cucumber/steps/dataservice/DataServiceSteps.java (hunk 2 above)
Comment (wjbuys) [reply integ-tests/src/cucumberTest/java/cucumber/steps/dataservice/DataServiceSteps.java:148]: Definitely, in a followup PR. Updating the cluster name actually makes no sense; we shouldn't support moving Environments between Clusters, or Weird Things May Happen.

File: integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java (hunk 4 above)
Comment (wjbuys) [reply integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java]: Whoops, yes, will fix.

File: integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java (hunk 1 above)
Comment (wjbuys) [reply integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java:79]: The other methods on this class are named for the type they create (compare `DescribeEnvironmentRequest describeEnvironmentRequest`), so I thought this naming would make sense in the context of this class; with an appropriately named variable, all invocations look something like:

```
create.describeEnvironmentRequest(...)
create.environmentId(...)
```

as opposed to:

```
creator.createDescribeEnvironmentRequest()
creator.createEnvironmentId()
```

Thoughts?

Comment (xjhe) [reply integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java:79]: Yes if we follow the pattern of the other methods, I think naming this as environmentId is fine.

Comment (wbingli) [reply integ-tests/src/cucumberTest/java/cucumber/steps/helpers/InputCreator.java:79]: I would prefer shorter naming,  `create.describeEnvironmentRequest`  looks good

**Review by wbingli — APPROVED** [review]
LGTM

**Pull request merged**